applications/serial_lte_modem/src/gnss/slm_at_gnss.c

```diff
@@ -0,0 +1,681 @@
+			if (interval == 0 || interval == 1 ||
+			   (interval >= 10 && interval <= 65535)) {
+				err = nrf_modem_gnss_fix_interval_set(interval);
+				if (err) {
+					LOG_ERR("Failed to set fix interval, error: %d", err);
+					return err;
+				}
+				/* start with connecting nRF Cloud */
+				err = cloud_connect(nrf_cloud);
+				if (err) {
+					LOG_ERR("Cloud connection failed, error: %d", err);
+				}
+			} else {
+				err = -EINVAL;
+			}
+		} else if (op == AGPS_STOP) {
+			err = cloud_disconnect(nrf_cloud);
+			if (err) {
+				LOG_ERR("Cloud disconnection failed, error: %d", err);
+			}
+			err = nrf_modem_gnss_stop();
+		} else {
+			err = -EINVAL;
+		} break;
+
+	case AT_CMD_TYPE_TEST_COMMAND:
+		sprintf(rsp_buf, "\r\n#XAGPS: (%d,%d),<interval>,<timeout>\r\n",
+			AGPS_STOP, AGPS_START);
+		rsp_send(rsp_buf, strlen(rsp_buf));
+		err = 0;
+		break;
+
+	default:
+		break;
+	}
+#endif /* CONFIG_SLM_AGPS */
+
+	return err;
+}
+
+/**@brief handle AT#XPGPS commands
+ *  AT#XPGPS=<op>[,<interval>]
+ *  AT#XPGPS? READ command not supported
+ *  AT#XPGPS=?
+ */
+int handle_at_pgps(enum at_cmd_type cmd_type)
+{
+	int err = -EINVAL;
+#if defined(CONFIG_SLM_PGPS)
+	uint16_t op;
+	uint16_t interval;
+
+	switch (cmd_type) {
+	case AT_CMD_TYPE_SET_COMMAND:
+		err = at_params_unsigned_short_get(&at_param_list, 1, &op);
+		if (err < 0) {
+			return err;
+		}
+		if (op == PGPS_START) {
+			err = at_params_unsigned_short_get(&at_param_list, 2, &interval);
+			if (err < 0) {
+				return err;
+			}
+			/* P-GPS is used in periodic mode only, in which timeout is ignored */
+			if (interval >= 10 && interval <= 65535) {
+				err = nrf_modem_gnss_fix_interval_set(interval);
+				if (err) {
+					LOG_ERR("Failed to set fix interval, error: %d", err);
+					return err;
+				}
+				/* start with connecting nRF Cloud */
+				err = cloud_connect(nrf_cloud);
+				if (err) {
+					LOG_ERR("Cloud connection failed, error: %d", err);
+				}
+			} else {
+				err = -EINVAL;
+			}
+		} else if (op == PGPS_STOP) {
+			err = cloud_disconnect(nrf_cloud);
+			if (err) {
+				LOG_ERR("Cloud disconnection failed, error: %d", err);
+			}
+			err = nrf_modem_gnss_stop();
+		} else {
+			err = -EINVAL;
+		} break;
+
+	case AT_CMD_TYPE_TEST_COMMAND:
+		sprintf(rsp_buf, "\r\n#XPGPS: (%d,%d),<interval>\r\n",
+			PGPS_STOP, PGPS_START);
+		rsp_send(rsp_buf, strlen(rsp_buf));
+		err = 0;
+		break;
+
+	default:
+		break;
+	}
+#endif /* CONFIG_SLM_PGPS */
+
+	return err;
+}
+
+/**@brief handle AT#XCELLPOS commands
+ *  AT#XCELLPOS=<op>
+ *  AT#XCELLPOS? READ command not supported
+ *  AT#XCELLPOS=?
+ */
+int handle_at_cellpos(enum at_cmd_type cmd_type)
+{
+	int err = -EINVAL;
+#if defined(CONFIG_SLM_CELL_POS)
+	uint16_t op;
+
+	switch (cmd_type) {
+	case AT_CMD_TYPE_SET_COMMAND:
+		err = at_params_unsigned_short_get(&at_param_list, 1, &op);
+		if (err < 0) {
+			return err;
+		}
+		if (op == CELLPOS_START_SCELL || op == CELLPOS_START_SCELL) {
+			if (op == CELLPOS_START_SCELL) {
```

Comment:
this is a lot of checks for `CELLPOS_START_SCELL` :)

Reply:
Embarrassing, one of them should be MCELL